Pflickin/refresh

pflickin opened the pull request:
- Updated the Python logging to use the new Opencensus azure packages.
- Removed uwsgi in favor of gunicorn.
- Updated the R base image to use a supported GDAL base.
- Several modifications to the ai4e_service and examples.
- Change python images to use guid task ids.
- Update R example and update SAS library for latest Azure Storage library.

yangsiyu007 commented on Containers/base-py/Dockerfile:
```diff
@@ -2,24 +2,48 @@
 
 # Install Azure Blob SDK
-RUN ./ai4e_api_tools/requirements/install-azure-blob.sh
+RUN pip3 install azure-mgmt-storage \
```
Should we pin the versions of the Azure packages or record them somewhere? Each time the base images are built, these versions may change.

yangsiyu007 commented on Examples/base-py/Dockerfile:
```diff
@@ -1,17 +1,15 @@
-
-# Example of installing additonal Anaconda packages - numpy and pandas
-RUN echo "source activate ai4e_py_api" >> ~/.bashrc \
-    && conda install -c conda-forge -n ai4e_py_api numpy pandas
```
Actually one more question - it'd be good to show how to install additional pip and conda packages. Is conda still available?

pflickin replied:
The API bases have been converted to use pip for all installs, so Conda is not needed.  I added a pip install example in this API.

pflickin commented on Containers/base-py/ai4e_api_tools/ai4e_app_insights.py:
```diff
@@ -102,28 +100,16 @@ def init_app(self, app, context):
+            self._appinsights_key = None
 
-        if self._key_grantee:
+        if self._appinsights_key:
```
This is the same app insights key that any user would use in production.  Comment added and changed output message.

pflickin commented on Containers/base-py/ai4e_api_tools/task_management/api_task.py:
```diff
@@ -44,10 +78,16 @@ def FailTask(self, taskId, status):
+        status['TaskId'] = taskId
+        status['Status'] = 'created'
+        status['Timestamp'] = datetime.datetime.strftime(datetime.datetime.now(), "%Y-%m-%d %H:%M:%S")
+        status['Endpoint'] = ''
```
The endpoint corresponds to the endpoint that is being tracked by the taskId.  It is only empty if there is an internal error.  I've changed the status from 'created' to 'Not found.'

pflickin commented on Containers/base-py/Dockerfile:
```diff
@@ -2,24 +2,48 @@
+        gunicorn
 
 # Install Azure Blob SDK
-RUN ./ai4e_api_tools/requirements/install-azure-blob.sh
```
Created a versioned requirements.txt pip file for each base image

pflickin commented:
- Updated the Python logging to use the new Opencensus azure packages.
- Removed uwsgi in favor of gunicorn.
- Updated the R base image to use a supported GDAL base.
- Change python images to use guid task ids.
- Several modifications to the ai4e_service and examples.
- Image version updates and documentation.
- Update R example and update SAS library for latest Azure Storage library.
- Removed separate app insights key requirement.
- Added versions to all pip packages within requirements.txt files.
- Resolved issues in the task manager (missing task, utc time, etc.)
